fem/bilininteg_hcurl.cpp

```diff
@@ -1950,12 +1950,7 @@ static void SmemPACurlCurlApply3D(const int D1D,
       } // qz
    }; // end of element loop
 
-   auto host_kernel = [&] MFEM_LAMBDA (int)
-   {
-      MFEM_ABORT_KERNEL("This kernel should only be used on GPU.");
-   };
-
-   ForallWrap<3>(true, NE, device_kernel, host_kernel, Q1D, Q1D, Q1D);
+   ForallWrap<3>(true, NE, device_kernel, Q1D, Q1D, Q1D);
```

Comment:
We should be careful here: I suspect this change will revert some fixes in #2788 for issue #2723. We need to find a new solution for the warnings.

Reply:
I reverted and allow back the `device` and `host` lambdas in [aa4b04d](https://github.com/mfem/mfem/pull/3454/commits/aa4b04d0a8711e5506191a212645ce26d8f9328a)

Reply:
This should work. Thanks!